packages/core/contracts/cross-chain-oracle/OracleBase.sol

```diff
@@ -71,11 +73,9 @@ abstract contract OracleBase {
     ) internal {
         bytes32 priceRequestId = _encodePriceRequest(identifier, time, ancillaryData);
         Price storage lookup = prices[priceRequestId];
-        if (lookup.state == RequestState.Requested) {
-            lookup.price = price;
-            lookup.state = RequestState.Resolved;
-            emit PushedPrice(identifier, time, ancillaryData, lookup.price, priceRequestId);
-        }
+        lookup.price = price;
+        lookup.state = RequestState.Resolved;
+        emit PushedPrice(identifier, time, ancillaryData, lookup.price, priceRequestId);
```

Comment:
I think there needs to be a bit more logic here:
1) if the `OracleBase` already has a price it should not change the stored price. 
2) we should only emit an event on the first time the function is called.

you can simply add the following to the top of this function:
```
if (lookup.state == RequestState.Resolved) return;
```

this way you exit early, don't override the price and don't double emit

Reply:
Agree with this comment.